src/node_options.cc

```diff
@@ -749,7 +749,6 @@ PerProcessOptionsParser::PerProcessOptionsParser(
             &PerProcessOptions::ssl_openssl_cert_store);
   Implies("--use-openssl-ca", "[ssl_openssl_cert_store]");
   ImpliesNot("--use-bundled-ca", "[ssl_openssl_cert_store]");
-#if NODE_FIPS_MODE
   AddOption("--enable-fips",
```

Comment:
I think the comment in the related issue

```
I did not really test the "vanilla" Node build, but from the above combinations
I assume that the --enable-fips option would not exist and without it,
the crypto.getFips() would still return 0, making this case equivalent to point 3.
```
is not correct.  This change means that the option will be available in the "vanilla" node builds, but that it will fail if you try to enable it.  That is likely the cause of a number of the test failures seen in the github actions runs.

@voxik, I thought the whole point was that the option would always be available for consistency across binaries, but would only work when either the binary was dynamically linked and on a system where FIPs was supported or at some point in the future if/when Node.js supports a FIPs mode again natively.  

If my understanding is correct there needs to be additional documentation explaining that, both so that its clear and so that the tests will pass.

Assuming my understanding is correct

Reply:
Yes, that is right. I think that the comment by @richardlau about `fipsMode` goes closer to the root cause, because there is apparently still enough functionality controlled by build options.